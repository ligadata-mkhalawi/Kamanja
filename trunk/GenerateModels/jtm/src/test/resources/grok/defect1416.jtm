{
  "header": {
    "namespace": "com.ligadata.jtm.test.grok",
    "version": "0.0.1",
    "language" : "scala",
    "minVersion" : "2.11"
  },

  "imports": {
    "packages": [ "com.ligadata.Utils._" ]
  },

  "aliases": {
    "messages": {
      "m1": "com.ligadata.kamanja.test.msg5",
      "o1": "com.ligadata.kamanja.test.msg2"
    }
  },

  "grok": {
    "name_grok_instance" : {
      "builtInDictionary": true,
      "match": "{EMAIL: email}",
      "file": ["path"],
      "patterns_dis": {
        "DOMAINTLD": "[a-zA-Z]+",
        "EMAIL": "%{NOTSPACE}@%{WORD}\\.%{DOMAINTLD}"
      }
    }
  },

  "transformations": {
    "test1": {
      "dependsOn": [ ["m1"] ],

      "grok match": {
        "in1": "%{IPORHOST:clientip} %{USER:ident} %{USER:auth} \\[%{HTTPDATE:timestamp}\\] \"(?:%{WORD:verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:httpversion})?|%{DATA:rawrequest})\" %{NUMBER:response} (?:%{NUMBER:bytes}|-)"
      },

      "outputs": {
        "o1": {
          "mapping": {
            "out1": "clientip",
            "out2": "ident",
            "out3": "auth",
            "out4": "verb"
          }
        }
      }
    }
  }
}
