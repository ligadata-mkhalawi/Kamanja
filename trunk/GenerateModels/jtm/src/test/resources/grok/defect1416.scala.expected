/*
* Copyright 2016 ligaDATA
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
package com.ligadata.jtm.test.grok.V1
import com.ligadata.KamanjaBase._
import com.ligadata.KvBase.TimeRange
import com.ligadata.kamanja.metadata.ModelDef
import com.ligadata.runtime._
import com.ligadata.Utils._
import org.aicer.grok.dictionary.GrokDictionary
// Package code start
// Package code end
class ModelFactory(modelDef: ModelDef, nodeContext: NodeContext) extends ModelInstanceFactory(modelDef, nodeContext) {
  // Factory code start
  // Factory code end
  override def createModelInstance(): ModelInstance = return new Model(this)
  override def getModelName: String = "com.ligadata.jtm.test.grok.Model"
  override def getVersion: String = "0.0.1"
  override def createResultObject(): ModelResultBase = new MappedModelResults()
  override def isModelInstanceReusable(): Boolean = true;
}
class common_exeGenerated_test1_1(conversion: com.ligadata.runtime.Conversion,
log: com.ligadata.runtime.Log,
context: com.ligadata.runtime.JtmContext,
msg1: com.ligadata.kamanja.test.V1000000.msg5) {
  import log._
  lazy val grok_instance_1_0_grp = grok_instance_1_0.extractNamedGroups(msg1.in1)
}
class common_exeGenerated_test1_1_process_o1(conversion: com.ligadata.runtime.Conversion,
log : com.ligadata.runtime.Log,
context: com.ligadata.runtime.JtmContext,
common: common_exeGenerated_test1_1,
msg1: com.ligadata.kamanja.test.V1000000.msg5) {
  import log._
  import common._
  val result: Array[MessageInterface]= try {
    val result = com.ligadata.kamanja.test.V1000000.msg2.createInstance
    result.out4 = (if(grok_instance_1_0_grp.containsKey("verb")) grok_instance_1_0_grp.get("verb") else "")
    result.out3 = (if(grok_instance_1_0_grp.containsKey("auth")) grok_instance_1_0_grp.get("auth") else "")
    result.out2 = (if(grok_instance_1_0_grp.containsKey("ident")) grok_instance_1_0_grp.get("ident") else "")
    result.out1 = (if(grok_instance_1_0_grp.containsKey("clientip")) grok_instance_1_0_grp.get("clientip") else "")
    if (result.hasTimePartitionInfo) result.setTimePartitionData ;
    if(context.CurrentErrors()==0) {
      Array(result)
    } else {
      Array.empty[MessageInterface]
    }
  } catch {
    case e: AbortOutputException => {
      context.AddError(e.getMessage)
      Array.empty[MessageInterface]
    }
    case e: Exception => {
      Debug("Exception: o1:" + e.getMessage)
      throw e
    }
  }
}
class Model(factory: ModelInstanceFactory) extends ModelInstance(factory) {
  val conversion = new com.ligadata.runtime.Conversion
  val log = new com.ligadata.runtime.Log(this.getClass.getName)
  val context = new com.ligadata.runtime.JtmContext
  import log._
  // Model code start
  // Model code end
  override def execute(txnCtxt: TransactionContext, execMsgsSet: Array[ContainerOrConcept], triggerdSetIndex: Int, outputDefault: Boolean): Array[ContainerOrConcept] = {
    context.Reset(); // Resetting the JtmContext before executing the model
    if (isTraceEnabled)
    Trace(s"Model::execute transid=%d triggeredset=%d outputdefault=%s".format(txnCtxt.transId, triggerdSetIndex, outputDefault.toString))
    if(isDebugEnabled)
    {
      execMsgsSet.foreach(m => Debug( s"Input: %s -> %s".format(m.getFullTypeName, m.toString())))
    }
    // Grok parts
    lazy val grok_instance_1: GrokDictionary = {
      val dict = new GrokDictionary
      dict.addBuiltInDictionaries
      dict.addDictionary(new File(getClass.getResource("grok/00346425/path").getPath))
      dict.bind()
      dict
    }
    lazy val grok_instance_1_0 = grok_instance_1.compileExpression("%{IPORHOST:clientip} %{USER:ident} %{USER:auth} \\[%{HTTPDATE:timestamp}\\] \"(?:%{WORD:verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:httpversion})?|%{DATA:rawrequest})\" %{NUMBER:response} (?:%{NUMBER:bytes}|-)")
    // Model methods
    def exeGenerated_test1_1(msg1: com.ligadata.kamanja.test.V1000000.msg5): Array[MessageInterface] = {
      Debug("exeGenerated_test1_1")
      context.SetSection("test1")
      val common = new common_exeGenerated_test1_1(conversion, log, context, msg1)
      def process_o1(): Array[MessageInterface] = {
        Debug("exeGenerated_test1_1::process_o1")
        context.SetScope("o1")
        val result = new common_exeGenerated_test1_1_process_o1(conversion, log, context, common, msg1)
        result.result
      }
      try {
        process_o1()
      } catch {
        case e: AbortTransformationException => {
          return Array.empty[MessageInterface]
        }
      }
    }
    // Evaluate messages
    val msgs = execMsgsSet.map(m => m.getFullTypeName -> m).toMap
    val msg1 = msgs.getOrElse("com.ligadata.kamanja.test.msg5", null).asInstanceOf[com.ligadata.kamanja.test.V1000000.msg5]
    // Main dependency -> execution check
    // Create result object
    val results: Array[MessageInterface] =
    try {
      (if(msg1!=null) {
        exeGenerated_test1_1(msg1)
      } else {
        Array.empty[MessageInterface]
      }) ++
      Array.empty[MessageInterface]
    } catch {
      case e: AbortExecuteException => {
        Array.empty[MessageInterface]
      }
    }
    if(isDebugEnabled)
    {
      results.foreach(m => Debug( s"Output: %s -> %s".format(m.getFullTypeName, m.toString())))
    }
    results.asInstanceOf[Array[ContainerOrConcept]]
  }
}
